client/src/components/body/profilepage/EnrolledStudentsTable.js

```diff
@@ -1,33 +1,116 @@
-/* eslint-disable react/prop-types */
-import React from "react";
-import { Table } from "reactstrap";
-import ResultLimiter from "./components/ResultLimiter";
+import React, { useEffect, useState } from "react";
+import StudentProgress from "./components/StudentProgress";
+import PropTypes from "prop-types";
+import { Input, Table } from "reactstrap";
+// import ResultLimiter from "./components/ResultLimiter";
 
 const EnrolledStudentsTable = (props) => {
-  const { assignedLabs, enrolledStudents, groupid } = props;
+  const { lab, enrolledStudents } = props;
+  const [search, setSearch] = useState("");
+
+  const displayAllStudents = () => {
+    return enrolledStudents.map((student, key) => {
+      return (
+        <StudentProgress
+          key={key}
+          student={student}
+          lab={lab}
+          hasLabel={true}
+          inTable={true}
+        />
+      );
+    });
+  };
+
+  const [displayedStudents, setDisplayedStudents] = useState(null);
+
+  useEffect(() => {
+    setDisplayedStudents(displayAllStudents);
+  }, []);
+
+  const searchStudents = (e) => {
+    e.preventDefault();
+    let searchResults = [];
+    enrolledStudents.forEach((student) => {
+      if (student.firstname.toLowerCase().includes(search.toLowerCase())) {
+        searchResults.push(student);
+      }
+    });
+    const displaySearchResults = searchResults.map((student, key) => {
+      return (
+        <StudentProgress
+          key={key}
+          student={student}
+          lab={lab}
+          hasLabel={true}
+          inTable={true}
+        />
+      );
+    });
+    if (searchResults.length > 0 && Array.isArray(searchResults)) {
+      setDisplayedStudents(displaySearchResults);
```

Comment:
does `displayedStudents` need to be a list of elements? I see this being the case in other components in this pr, where we use `useState` to store a list of components. I would say we should steer away from that because we could end up in weird states when props become stale. alternatively, we should use `useState` to store the actual data that describes a component. once we have a list of objects with the data, the list of components/elements should be created during render

Reply:
Fixed this and no longed splits everything up. Also added search error handling.